pkg/cmd/chat/push/push.go

```diff
@@ -0,0 +1,154 @@
+package push
+
+import (
+	"encoding/json"
+
+	stream_chat "github.com/GetStream/stream-chat-go/v5"
+	"github.com/GetStream/stream-cli/pkg/config"
+	"github.com/GetStream/stream-cli/pkg/utils"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/spf13/cobra"
+)
+
+func NewCmds() []*cobra.Command {
+	return []*cobra.Command{
+		updateCmd(),
+		testCmd(),
+	}
+}
+
+func updateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "upsert-pushprovider --properties [raw-json]",
+		Short: "Create or updates a push provider",
+		Long: `
+			The "--properties" parameter expects a raw json string that can be
+			unmarshalled into a stream_chat.PushProvider object on the Go SDK side.
+			See the example section.
+			Available properties:
+			type
+			name
+			description
+			disabled_at
+			disabled_reason
+					
+			apn_auth_key
+			apn_key_id
+			apn_team_id
+			apn_topic
+					
+			firebase_credentials
+					
+			huawei_app_id
+			huawei_app_secret
+					
+			xiaomi_package_name
+			xiaomi_app_secret
+		`,
+		Example: heredoc.Doc(`
+			# Setting up an APN push provider
+			$ stream-cli chat upsert-pushprovider --properties "{'type': 'apn', 'name': 'staging', 'apn_auth_key': 'key', 'apn_key_id': 'id', 'apn_topic': 'topic', 'apn_team_id': 'id'}"
+
+			# Setting up a Firebase push provider
+			$ stream-cli chat upsert-pushprovider --properties "{'type': 'firebase', 'name': 'staging', 'firebase_credentials': 'credentials'}"
+
+			# Setting up a Huawei push provider
+			$ stream-cli chat upsert-pushprovider --properties "{'type': 'huawei', 'name': 'staging', 'huawei_app_id': 'id', 'huawei_app_secret': 'secret'}"
+
+			# Setting up a Xiaomi push provider
+			$ stream-cli chat upsert-pushprovider --properties "{'type': 'xiaomi', 'name': 'staging', 'xiaomi_package_name': 'name', 'xiaomi_app_secret': 'secret'}"
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			prop, _ := cmd.Flags().GetString("properties")
+
+			var p stream_chat.PushProvider
+			err = json.Unmarshal([]byte(prop), &p)
+			if err != nil {
+				return err
+			}
+
+			_, err = c.UpsertPushProvider(cmd.Context(), &p)
+			if err != nil {
+				return err
+			}
+
+			cmd.Println("Successfully updated push provider.")
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("properties", "p", "", "[required] Raw json properties to send to the backend")
+	cmd.MarkFlagRequired("properties")
+
+	return cmd
+}
+
+func testCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "test-push --message-id [string] --apn-template [string]" +
```

Comment:
apn-template, firebase-template, firebase-data-temlate are V1 only. We're going to add support for templates in V2 but this is a different thing.

Reply:
I kinda followed this: https://github.com/GetStream/chat/blob/9ef8bb9ad17034cd902d256f23c653b9d89c6cc4/controllers/check_push.go#L82

Reply:
Let's add theses fields only to the upsert-push-provider command (add to heredoc). As for `check_push` endpoint we need to support it on backend first.

Reply:
@driver-devel okay, but I'll need to add it to the Go SDK first. one minute

Reply:
> firebase_apn_notification_template

it's call `firebase_apn_template` actually

> firebase_notification_template

we will keep this but good to drop data one `firebase_data_template`

Reply:
https://github.com/GetStream/stream-chat-go/pull/218